File: pec-observation-app/android-app/app/src/main/kotlin/br/gov/educacao/sp/pecobservation/ports/AudioRecordingPort.kt
package br.gov.educacao.sp.pecobservation.ports

import br.gov.educacao.sp.pecobservation.domain.entities.AudioRecordingResult
import kotlinx.coroutines.flow.StateFlow
import java.util.UUID

enum class RecordingState {
    IDLE, PREPARING, RECORDING, PAUSED, STOPPING, COMPLETED, FAILED
}

interface AudioRecordingPort {
    val state: StateFlow<RecordingState>
    val elapsedSeconds: StateFlow<Int>

    suspend fun requestPermission(): Boolean
    suspend fun startRecording(observationId: UUID)
    suspend fun pauseRecording()
    suspend fun resumeRecording()
    suspend fun stopRecording(): AudioRecordingResult
    fun cancelRecording()
}

File: pec-observation-app/ios-app/Sources/PECObservation/Ports/AudioRecordingPort.swift
import Foundation

public enum RecordingState: String, Equatable {
    case idle
    case preparing
    case recording
    case paused
    case stopping
    case completed
    case failed
}

// Exact protocol from SDD section 13.3
public protocol AudioRecordingPort {
    func requestPermission() async -> Bool
    func startRecording(observationId: UUID) async throws
    func pauseRecording() async throws
    func resumeRecording() async throws
    func stopRecording() async throws -> AudioRecordingResult
    var currentState: RecordingState { get }
    var elapsedSeconds: Int { get }
}
